File: src/main/java/mandarin/packpack/commands/bot/Memory.java
package mandarin.packpack.commands.bot;

import common.CommonStatic;
import mandarin.packpack.commands.ConstraintCommand;
import mandarin.packpack.supporter.server.CommandLoader;
import mandarin.packpack.supporter.server.data.IDHolder;
import net.dv8tion.jda.api.entities.channel.middleman.MessageChannel;

import javax.annotation.Nonnull;

public class Memory extends ConstraintCommand {
    public Memory(ROLE role, CommonStatic.Lang.Locale lang, IDHolder id) {
        super(role, lang, id, false);
    }

    @Override
    public void doSomething(@Nonnull CommandLoader loader) {
        MessageChannel ch = loader.getChannel();

        long f = Runtime.getRuntime().freeMemory();
        long t = Runtime.getRuntime().totalMemory();
        long m = Runtime.getRuntime().maxMemory();
        double per = 100.0 * (t - f) / m;

        ch.sendMessage("Memory used: " + (t - f >> 20) + " MB / " + (m >> 20) + " MB, " + (int) per + "%").queue();
    }
}

File: src/main/java/mandarin/card/commands/Memory.kt
package mandarin.card.commands

import common.CommonStatic
import mandarin.packpack.commands.Command
import mandarin.packpack.supporter.bc.DataToString
import mandarin.packpack.supporter.server.CommandLoader
import kotlin.math.round

class Memory : Command(CommonStatic.Lang.Locale.EN, false) {
    override fun doSomething(loader: CommandLoader) {
        val f = Runtime.getRuntime().freeMemory()
        val t = Runtime.getRuntime().totalMemory()
        val m = Runtime.getRuntime().maxMemory()

        val percentage = 100.0 * (t - f) / m

        replyToMessageSafely(loader.channel, "Memory Used : ${t - f shr 20} MB / ${m shr 20} MB, ${DataToString.df.format(percentage)}%", loader.message) { a -> a }
    }
}